planner/core/exhaust_physical_plans.go

```diff
@@ -2227,20 +2227,73 @@ func (la *LogicalAggregation) getStreamAggs(prop *property.PhysicalProperty) []P
 	return streamAggs
 }
 
+// TODO: support more operators and distinct later
+func (la *LogicalAggregation) checkCanPushDownToMPP() bool {
+	for _, agg := range la.AggFuncs {
+		if agg.HasDistinct {
+			return false
+		}
+		if agg.Name != ast.AggFuncSum && agg.Name != ast.AggFuncMin && agg.Name != ast.AggFuncCount && agg.Name != ast.AggFuncMax && agg.Name != ast.AggFuncFirstRow && agg.Name != ast.AggFuncAvg {
+			return false
+		}
+		if !expression.CanExprsPushDown(la.SCtx().GetSessionVars().StmtCtx, agg.Args, la.SCtx().GetClient(), kv.TiFlash) {
+			return false
+		}
+	}
+	return true
+}
+
+func (la *LogicalAggregation) getHashAggsForMPP(prop *property.PhysicalProperty) (hashAggs []PhysicalPlan) {
+	partitionCols := la.GetGroupByCols()
+	if len(partitionCols) != 0 {
```

Comment:
The design braches should be 
- if the type is mpp, check if it is match partition cols if it's a hash type, if it is a bcj type I think we must deny it.
- if the type is root, try tidb-mpp fusion mode only if mpp mode is enabled.

Reply:
you are right.
as well the hash join should do 1?

Reply:
hash join can implement a root task property.
Actually for agg we can also do a 1-phase, 2-phase and fusion agg when encountering root task.

Reply:
I mean hashjoin should also reject `broadcast`.

Reply:
yeah you are right, you can add it.

Reply:
added.